### PR TITLE
components: forbid visibility changes if it contains incoherent records

### LINKED ISSUE #818: public record in restricted community
Package version (if known): invenio-app-rdm v10? (https://inveniordm.web.cern.ch 2022-10-17)

## Describe the bug
I was able to have a public record in a restricted community.

## Steps to Reproduce
1. Create a "public" community.
2. Publish a fully "public" record within this community.
3. Change the community visibility to "restricted"
4. See public record within a restricted community

## Expected behavior

As described here:
https://discord.com/channels/692989811736182844/704625518552547329/1030462802491617312
if a community contains public records the community cannot be changed to restricted 

## Screenshots (if applicable)

image
image

### PR DESCRIPTION
**Pull request by ppanero**
- closes #818 

Notes:
- The code is not functional, just pseudocode to decide if the solution is accepted.
- I am assuming that we should not allow to:
   - change from public to restricted, if there are public records
   - change from restricted to public, if there are restricted records (this could be ok though 🤔 )

### CODE REVIEW

File: invenio_communities/communities/services/components.py [hunk 1]
```diff
@@ -79,6 +79,18 @@ def create(self, identity, data=None, record=None, **kwargs):
     def update(self, identity, data=None, record=None, **kwargs):
         """Update handler."""
         self._populate_access_and_validate(identity, data, record, **kwargs)
+        visibility = data["access"]["visibility"]
+        opposite = "public" if visibility == "public" else "restricted"
```
Comment (ppanero) [inline invenio_communities/communities/services/components.py:83]: I do not like hardcoding this here, maybe could be delegated to the access sysfield, although `record.access.opposite_visibility`sounds quite ugly

**Comment by ntarocco** [comment]
> change from restricted to public, if there are restricted records (this could be ok though 🤔 )

[Current implementation](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/review/service.py#L203):
- community restricted? record must be restricted
- community public? record can be restricted or public

File: invenio_communities/communities/services/components.py (hunk 1 above)
Comment (ntarocco) [reply invenio_communities/communities/services/components.py:83]: See the implementation I linked in the other comment: I ended up adding an `assert` to make sure.

**Comment by ntarocco** [comment]
I hope my code helps. You will need another change in `rdm-records` probably, when the record restriction is changed from restricted to public: if it is in at least one community that is restricted, you cannot change it.

**Comment by ppanero** [comment]
@ntarocco thanks a lot for the code snippets. I will proceed with the implementation (record agnostic here, and then with records knowledge in rdm-records)

**Comment by ppanero** [comment]
closed in favour of https://github.com/inveniosoftware/invenio-rdm-records/pull/1236

**Pull request closed without merging**